### ISSUE TITLE
Enhacement: Remove the 'Class' filter in the 'TEntity' generic type-argument for the 'ExecuteQuery' operation.

### ISSUE THREAD

**Issue by mikependon** [mikependon/RepoDB#549]
To further support the case of the #538, we need to eliminate the filter of the `class` in the [ExecuteQuery](https://repodb.net/operation/executequery) operation.

**From:**

```csharp
public static IEnumerable<TEntity> ExecuteQuery<TEntity>(this IDbConnection connection,
	string commandText,
	object param = null,
	CommandType? commandType = null,
	string cacheKey = null,
	int? cacheItemExpiration = Constant.DefaultCacheItemExpirationInMinutes,
	int? commandTimeout = null,
	IDbTransaction transaction = null,
	ICache cache = null)
	where TEntity : class
{
	return ExecuteQueryInternal<TEntity>(connection: connection,
		commandText: commandText,
		param: param,
		commandType: commandType,
		cacheKey: cacheKey,
		cacheItemExpiration: cacheItemExpiration,
		commandTimeout: commandTimeout,
		transaction: transaction,
		cache: cache,
		skipCommandArrayParametersCheck: false);
}

```

**To:**

```csharp
public static IEnumerable<TResult> ExecuteQuery<TResult>(this IDbConnection connection,
	string commandText,
	object param = null,
	CommandType? commandType = null,
	string cacheKey = null,
	int? cacheItemExpiration = Constant.DefaultCacheItemExpirationInMinutes,
	int? commandTimeout = null,
	IDbTransaction transaction = null,
	ICache cache = null)
{
	return ExecuteQueryInternal<TResult>(connection: connection,
		commandText: commandText,
		param: param,
		commandType: commandType,
		cacheKey: cacheKey,
		cacheItemExpiration: cacheItemExpiration,
		commandTimeout: commandTimeout,
		transaction: transaction,
		cache: cache,
		skipCommandArrayParametersCheck: false);
}
```

**Comment by mikependon** [comment]
The fix is available at version [RepoDb v1.12.0-beta3](https://www.nuget.org/packages/RepoDb/1.12.0-beta3) or above.

**Comment by mikependon** [comment]
Closing this incident ticket now.
